packages/telegram-stamper/src/index.ts

```diff
@@ -0,0 +1,185 @@
+  stamper: ApiKeyStamper | undefined;
+  cloudStorageKeySuffix: string = "";
+
+  constructor(config?: TTelegramStamperConfig) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check the type of config that was passed in to the constructor
+    if (config) {
+      // if a public/private api key was passed in instantiate the stamper
+      this.stamper = new ApiKeyStamper({
+        apiPublicKey: config.apiPublicKey,
+        apiPrivateKey: config.apiPrivateKey,
+      });
+
+      // set the key cloud storage key suffix
+      if (config.cloudStorageKeySuffix) {
+        // add an underscore prior to the suffix so the key is readable
+        this.cloudStorageKeySuffix = "_" + config.cloudStorageKeySuffix;
+      }
+    }
+  }
+
+  // initialize the telegram stamper by getting/setting the private/public api key values from/to telegram cloud storage
+  async init() {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    if (this.stamper) {
+      // insert creds into telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.setItem(
+        this.getCloudStorageKeyName(),
+        this.stringifyAPIKey(),
+        (err: any, stored: boolean) => {
+          if (err != null || !stored) {
+            throw new TelegramStamperError(
+              `Failed inserting api key into Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+        }
+      );
+    } else {
+      // attempt to get creds from telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.getItem(
+        this.getCloudStorageKeyName(),
+        (err: any, apiKey: string) => {
+          if (err != null || !apiKey) {
+            throw new TelegramStamperError(
+              `Failed getting api key from Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+
+          const { parsedPublicKey, parsedPrivateKey } =
+            this.parseAPIKey(apiKey);
+
+          // ToDo: api key validation, how does apikeystamper validate creds passed to it?'
+
+          if (!parsedPublicKey || !parsedPrivateKey) {
+            throw new TelegramStamperError(
+              "Failed parsing api key from Telegram Cloud Storage"
+            );
+          }
+
+          // instantiate api key stamper
+          this.stamper = new ApiKeyStamper({
+            apiPublicKey: parsedPublicKey,
+            apiPrivateKey: parsedPrivateKey,
+          });
+        }
+      );
+    }
+  }
+
+  async stamp(payload: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramStamperError(
+        "Cannot stamp with unintialized telegram stamper, call TelegramStamper.init()"
+      );
+    }
```

Comment:
this could be axed if we had the `.create()` pattern, potentially! It's pretty easy to imagine devs forgetting to call `.init()`. Not the worst thing but kind of annoying (it's one more thing to remember)

Reply:
changed to this pattern :)